velox/exec/HashProbe.cpp

```diff
@@ -1881,82 +1882,7 @@ void HashProbe::maybeSetupSpillOutputReader() {
   spillOutputPartitionSet_.clear();
 }
 
-SpillPartitionSet HashProbe::spillTable() {
-  struct SpillResult {
-    std::unique_ptr<Spiller> spiller{nullptr};
-    const std::exception_ptr error{nullptr};
-
-    explicit SpillResult(std::exception_ptr _error) : error(_error) {}
-    explicit SpillResult(std::unique_ptr<Spiller> _spiller)
-        : spiller(std::move(_spiller)) {}
-  };
-
-  const std::vector<RowContainer*> rowContainers = table_->allRows();
-  std::vector<std::shared_ptr<AsyncSource<SpillResult>>> spillTasks;
-  auto* spillExecutor = spillConfig()->executor;
-  for (auto* rowContainer : rowContainers) {
-    if (rowContainer->numRows() == 0) {
-      continue;
-    }
-    spillTasks.push_back(memory::createAsyncMemoryReclaimTask<SpillResult>(
-        [this, rowContainer]() {
-          try {
-            return std::make_unique<SpillResult>(spillTable(rowContainer));
-          } catch (const std::exception& e) {
-            LOG(ERROR) << "Spill sub-table from hash probe pool "
-                       << pool()->name() << " failed: " << e.what();
-            // The exception is captured and thrown by the caller.
-            return std::make_unique<SpillResult>(std::current_exception());
-          }
-        }));
-    if ((spillTasks.size() > 1) && (spillExecutor != nullptr)) {
-      spillExecutor->add([source = spillTasks.back()]() { source->prepare(); });
-    }
-  }
-
-  auto syncGuard = folly::makeGuard([&]() {
-    for (auto& spillTask : spillTasks) {
-      // We consume the result for the pending tasks. This is a cleanup in the
-      // guard and must not throw. The first error is already captured before
-      // this runs.
-      try {
-        spillTask->move();
-      } catch (const std::exception&) {
-      }
-    }
-  });
-
-  SpillPartitionSet spillPartitions;
-  for (auto& spillTask : spillTasks) {
-    const auto result = spillTask->move();
-    if (result->error) {
-      std::rethrow_exception(result->error);
-    }
-    result->spiller->finishSpill(spillPartitions);
-  }
-
-  // Remove the spilled partitions which are empty so as we don't need to
-  // trigger unnecessary spilling at hash probe side.
-  removeEmptyPartitions(spillPartitions);
-  return spillPartitions;
-}
-
-std::unique_ptr<Spiller> HashProbe::spillTable(RowContainer* subTableRows) {
-  VELOX_CHECK_NOT_NULL(tableSpillType_);
-
-  auto tableSpiller = std::make_unique<Spiller>(
-      Spiller::Type::kHashJoinBuild,
-      joinType_,
-      subTableRows,
-      tableSpillType_,
-      std::move(tableSpillHashBits_),
-      spillConfig(),
-      &spillStats_);
-  tableSpiller->spill();
-  return tableSpiller;
-}
-
-void HashProbe::prepareTableSpill(
+void HashProbe::checkMaxSpillLevel(
```

Comment:
Let's keep calling this prepareTableSpill? Thanks!

Reply:
The main theme of this method after refactoring is max spill level?